pypesto/objective/petab_import.py

```diff
@@ -0,0 +1,262 @@
+import numpy as np
+import pandas as pd
+import os
+import sys
+import importlib
+import numbers
+
+import amici
+import petab
+from pypesto.objective import AmiciObjective
+from pypesto.problem import Problem
+
+
+class PetabImporter:
+
+    def __init__(self, petab_problem, output_folder=None, force_compile=False):
+        """
+        petab_problem: petab.Problem
+            Managing access to the model and data.
+
+        output_folder: str,  optional
+            Folder to contain the amici model. Defaults to
+            './tmp/petab_problem.name'.
+
+        force_compile: str, optional
+            If False, the model is compiled only if the output folder does not
+            exist yet. If True, the output folder is deleted and the model
+            (re-)compiled in either case.
+        """
+        self.petab_problem = petab_problem
+
+        if output_folder is None:
+            output_folder = os.path.abspath(
+                os.path.join("tmp", self.petab_problem.name))
+        self.output_folder = output_folder
+
+        self.model = None
+        self.import_model(force_compile)
+
+        self.solver = self.model.getSolver()
+
+    @staticmethod
+    def from_folder(folder, output_folder=None, force_compile=False):
+        """
+        Simplified constructor exploiting the standardized petab folder
+        structure.
+
+        Parameters
+        ----------
+
+        folder: str
+            Path to the base folder of the model, as in
+            petab.Problem.from_folder.
+
+        output_folder, force_compile: see __init__.
+        """
+        petab_problem = petab.Problem.from_folder(folder)
+
+        return PetabImporter(
+            petab_problem=petab_problem,
+            output_folder=output_folder,
+            force_compile=force_compile
+        )
+
+    def import_model(self, force_compile=False):
+        """
+        Import amici model. If necessary or force_compile is True, compile
+        first.
+        """
+        # compile
+        if not os.path.exists(self.output_folder) or force_compile:
+            self.compile_model()
+
+        # add module to path
+        if self.output_folder not in sys.path:
+            sys.path.insert(0, self.output_folder)
+
+        # load moduĺe
+        model_module = importlib.import_module(self.petab_problem.name)
+
+        # import model
+        self.model = model_module.getModel()
+
+    def compile_model(self):
```

Comment:
Shall we move this one to AMICI? Might be easier to keep it in sync with the code it depends on and may be more reusable.

Reply:
definitely an option. why i didn't so far:
1) i did not want to split up the funtcionality into too many places (petab, pypesto.petab_import, pypesto.amici_objective, amici ...)
2) the comiplation-specific part is rather short, then already the objective creation begins